### PR TITLE
:bug: Fix Intercom stream base url in check connection

### LINKED ISSUE #11177: Source Intercom: extra slash in Intercom base url
## Environment
- **Airbyte version**:  0.35.55-alpha
- **OS Version / Instance**: Ubuntu 20.04
- **Deployment**: Docker compose
- **Source Connector and version**: Intercom 0.1.13
- **Severity**: Medium
- **Step where error happened**: Setup new connection

## Current Behavior
Unable to test the connection

## Expected Behavior
Being able to test the connection

## Logs
*If applicable, please upload the logs from the failing operation. 
For sync jobs, you can download the full logs from the UI by going to the sync attempt page and 
clicking the download logs button at the top right of the logs display window.*

<details>
<summary>LOG</summary> 

```
2022-03-15 19:30:35 INFO i.a.w.t.TemporalAttemptExecution(get):105 - Docker volume job log path: /tmp/workspace/849ce07d-1ad2-4664-87e1-35ed5e28a78b/0/logs.log
2022-03-15 19:30:35 INFO i.a.w.t.TemporalAttemptExecution(get):110 - Executing worker wrapper. Airbyte version: 0.35.55-alpha
2022-03-15 19:30:35 INFO i.a.c.i.LineGobbler(voidCall):82 - Checking if airbyte/source-intercom:0.1.13 exists...
2022-03-15 19:30:35 INFO i.a.w.p.DockerProcessFactory(create):106 - Creating docker job ID: 849ce07d-1ad2-4664-87e1-35ed5e28a78b
2022-03-15 19:30:35 INFO i.a.c.i.LineGobbler(voidCall):82 - airbyte/source-intercom:0.1.13 was found locally.
2022-03-15 19:30:35 INFO i.a.w.p.DockerProcessFactory(create):158 - Preparing command: docker run --rm --init -i -w /data/849ce07d-1ad2-4664-87e1-35ed5e28a78b/0 --log-driver none --network host -v airbyte_workspace:/data -v /tmp/airbyte_local:/local -e WORKER_JOB_ATTEMPT=0 -e WORKER_CONNECTOR_IMAGE=airbyte/source-intercom:0.1.13 -e AIRBYTE_ROLE= -e WORKER_ENVIRONMENT=DOCKER -e AIRBYTE_VERSION=0.35.55-alpha -e WORKER_JOB_ID=849ce07d-1ad2-4664-87e1-35ed5e28a78b airbyte/source-intercom:0.1.13 check --config source_config.json
2022-03-15 19:30:37 WARN c.n.s.JsonMetaSchema(newValidator):338 - Unknown keyword existingJavaType - you should define your own Meta Schema. If the keyword is irrelevant for validation, just use a NonValidationKeyword
2022-03-15 19:30:37 ERROR i.a.w.p.a.DefaultAirbyteStreamFactory(internalLog):95 - Check failed
2022-03-15 19:30:37 INFO i.a.w.t.TemporalAttemptExecution(get):131 - Stopping cancellation check scheduling...
```

</details>

## Steps to Reproduce
1. Create new source
2. Choose Intercom
3. Put in your access token and test

## Are you willing to submit a PR?
Already done #11176

### PR DESCRIPTION
**Pull request by mrhallak**
## What
Hello Airbyte team! I was giving the tool a shot and wanted to use Intercom as a source. I noticed the extra slash at the end of the URL through the logs I got because I was not able to connect. 

## How
The solution is pretty simple - just removing the slash at the end of the base url.

## Recommended reading order
1. `python`

## 🚨 User Impact 🚨
There shouldn't be any breaking changes here

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests
Note some tests failed - it seems the Intercom token I have does not grant me access to everything which is expected because I'm using a token provided by my organization

<details><summary><strong>Acceptance</strong></summary>

Results (17.32s):
      17 passed
       5 failed
         - ../airbyte/source_acceptance_test/source_acceptance_test/tests/test_core.py:144 TestConnection.test_check[inputs0]
         - ../airbyte/source_acceptance_test/source_acceptance_test/tests/test_core.py:355 TestBasicRead.test_read[inputs0]
         - ../airbyte/source_acceptance_test/source_acceptance_test/tests/test_full_refresh.py:39 TestFullRefresh.test_sequential_reads[inputs0]
         - ../airbyte/source_acceptance_test/source_acceptance_test/tests/test_incremental.py:81 TestIncremental.test_two_sequential_reads[inputs0]
         - ../airbyte/source_acceptance_test/source_acceptance_test/tests/test_incremental.py:105 TestIncremental.test_state_with_abnormally_large_values[inputs0]

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-intercom/source_intercom/source.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 
 class IntercomStream(HttpStream, ABC):
-    url_base = "https://api.intercom.io/"
+    url_base = "https://api.intercom.io"
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-intercom/source_intercom/source.py:21]: this is concatenated with stream paths below to form `https://api.intercom.io/<path>` -- is it safe to do this?

Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-intercom/source_intercom/source.py:21]: Doesn't matter `urljoin` will parse correctly with `/` or without. 

The problem happens in check connection operation where is concatenating using f-string and not urljoin.
https://github.com/airbytehq/airbyte/blob/266a491c8e3236b7e6603f0c34aa7ac1abf0a4c3/airbyte-integrations/connectors/source-intercom/source_intercom/source.py#L416-L417

**Comment by marcosmarxm** [comment]
@mrhallak can you sign the CLA?

**Comment by mrhallak** [comment]
> @mrhallak can you sign the CLA?

Done!

File: airbyte-integrations/connectors/source-intercom/source_intercom/source.py (hunk 1 above)
Comment (mrhallak) [reply airbyte-integrations/connectors/source-intercom/source_intercom/source.py:21]: @sherifnada @marcosmarxm Would you guys prefer it if I change it to a `urljoin`?

Comment (sherifnada) [reply airbyte-integrations/connectors/source-intercom/source_intercom/source.py:21]: That would be better imo

Comment (mrhallak) [reply airbyte-integrations/connectors/source-intercom/source_intercom/source.py]: Updated!

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**